### PR TITLE
Add more Types needed for existing and next upcoming clusters

### PR DESCRIPTION
**Pull request by Apollon77**
I adjusted two types and added several more needed by my next upcoming PRs :-)

### CODE REVIEW

File: src/matter/common/AttributeId.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+/**
+ * An Attribute ID is a 32 bit number and indicates an attribute defined in a cluster specification.
+ *
+ * @see {@link MatterCoreSpecificationV1_0} § 7.18.2.15
+ */
+export type AttributeId = { attributeId: true /* Hack to force strong type checking at compile time */ };
+export const AttributeId = (id: number) => id as unknown as AttributeId;
+
+/** Data model for a Vendor Identifier. */
```
Comment (mfucci) [inline src/matter/common/AttributeId.ts:18]: Attribute ID

File: src/matter/common/ClusterId.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+/**
+ * A Cluster Identifier is a 32 bit number and SHALL reference a single cluster specification and
+ * SHALL define conformance to that specification.
+ *
+ * @see {@link MatterCoreSpecificationV1_0} § 7.10
+ */
+export type ClusterId = { clusterId: true /* Hack to force strong type checking at compile time */ };
+export const ClusterId = (id: number) => id as unknown as ClusterId;
+
+/** Data model for a Vendor Identifier. */
```
Comment (mfucci) [inline src/matter/common/ClusterId.ts:19]: Cluster Identifier

File: src/matter/common/DeviceTypeId.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+/**
+ * A Device type ID is a 32-bit number that defines the type of the device
+ *
+ * @see {@link MatterCoreSpecificationV1_0} § 7.15
+ */
+export type DeviceTypeId = { deviceTypeId: true /* Hack to force strong type checking at compile time */ };
+export const DeviceTypeId = (id: number) => id as unknown as DeviceTypeId;
+
+/** Data model for a Vendor Identifier. */
```
Comment (mfucci) [inline src/matter/common/DeviceTypeId.ts:18]: Device type ID

File: src/matter/common/EndpointNumber.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+/**
+ * A Endpoint Number is a 16-bit number that that indicates an instance of a device type.
+ *
+ * @see {@link MatterCoreSpecificationV1_0} § 7.18.2.11
+ */
+export type EndpointNumber = { endpointNumber: true /* Hack to force strong type checking at compile time */ };
+export const EndpointNumber = (id: number) => id as unknown as EndpointNumber;
+
+/** Data model for a Vendor Identifier. */
```
Comment (mfucci) [inline src/matter/common/EndpointNumber.ts:18]: Endpoint number

File: src/matter/common/FabricId.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+/**
+ * A Fabric ID is a 64-bit number that uniquely identifies the Fabric within the scope of
+ * a particular root CA
+ *
+ * @see {@link MatterCoreSpecificationV1_0} § 2.5.1
+ */
+export type FabricId = { fabricId: true /* Hack to force strong type checking at compile time */ };
+export const FabricId = (id: bigint) => id as unknown as FabricId;
+
+/** Explicitly convert a NodeId to a bigint */
```
Comment (mfucci) [inline src/matter/common/FabricId.ts:19]: Fabric ID

File: src/matter/common/FabricId.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+export const nodeIdToBigint = (fabricId: FabricId) => fabricId as unknown as bigint;
+
+/** Data model for a Node Identifier. */
```
Comment (mfucci) [inline src/matter/common/FabricId.ts:22]: Fabric ID

File: src/matter/common/FabricId.ts [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+ * Copyright 2022 The node-matter Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Typed, UnsignedLongT } from "../../codec/TlvObjectCodec";
+import { MatterCoreSpecificationV1_0 } from "../../Specifications";
+
+/**
+ * A Fabric ID is a 64-bit number that uniquely identifies the Fabric within the scope of
+ * a particular root CA
```
Comment (mfucci) [inline src/matter/common/FabricId.ts:12]: Missing .

File: src/matter/common/DeviceTypeId.ts [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+ * @license
+ * Copyright 2022 The node-matter Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Typed, UnsignedIntT } from "../../codec/TlvObjectCodec";
+import { MatterCoreSpecificationV1_0 } from "../../Specifications";
+
+/**
+ * A Device type ID is a 32-bit number that defines the type of the device
```
Comment (mfucci) [inline src/matter/common/DeviceTypeId.ts:11]: Missing .

File: src/matter/common/GroupId.ts [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+ * 0x0000 - Null or unspecified Group ID
+ * 0x0001 - 0xFEFF: Application Group ID, assigned by fabric Administrator
+ * 0xFF00 - 0xFFFF: Universal Group ID range reserved for static multicast and anycast identifiers
+ *
+ * @see {@link MatterCoreSpecificationV1_0} § 2.5.4
+ */
+export type GroupId = { groupId: true /* Hack to force strong type checking at compile time */ };
+export const GroupId = (id: number) => id as unknown as GroupId;
+
+/** Explicitly convert a GroupId to a number */
```
Comment (mfucci) [inline src/matter/common/GroupId.ts:26]: convert -> converts

File: src/matter/common/GroupId.ts [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+export const groupIdToNumber = (groupId: GroupId) => groupId as unknown as number;
+
+/** Data model for a Vendor Identifier. */
```
Comment (mfucci) [inline src/matter/common/GroupId.ts:29]: Group ID

File: src/matter/common/VendorId.ts [hunk 11]
```diff
@@ -4,18 +4,18 @@
...
+ * Connectivity Standards Alliance (see [CSA Manufacturer Code Database]).
+ *
  * @see {@link MatterCoreSpecificationV1_0} § 2.5.2
  */
 export type VendorId = { vendorId: true /* Hack to force strong type checking at compile time */ };
 export const VendorId = (id: number) => id as unknown as VendorId;
 
 /** Data model for a Vendor Identifier. */
-export const VendorIdT = Typed<VendorId>(UnsignedIntT);
+export const VendorIdT = Typed<VendorId>(BoundedUnsignedIntT({ min: 0, max: 0xFFFF }));
```
Comment (mfucci) [inline src/matter/common/VendorId.ts:21]: I will add a PR to introduce UInt16T. Maybe you can merge and use it here

**Review by mfucci — APPROVED** [review]
Minor comments about mistakes in the doc.
I will probably refactor those to avoid use the type hack but fine to commit them since it should not change the interface.

File: src/matter/common/AttributeId.ts (hunk 1 above)
Comment (Apollon77) [reply src/matter/common/AttributeId.ts:18]: meeh ... copy paste :-) fixing

File: src/matter/common/VendorId.ts (hunk 11 above)
Comment (Apollon77) [reply src/matter/common/VendorId.ts:21]: Because PR not ready as of your info it is slightly delayed ... :-)

**Comment by Apollon77** [comment]
Sorry for the copy paste errors ... I think my brain was too strange after these hours reading specs ;-)

FIxed ... For the Unbound: As proposed two options. I adjust after you finished your PR and we merged it. Or we adjust your PR that old types are still in and I "convert" - and then we kill the old ones. Would allow a non-blocking working - for me both ok.

I have alreydy type adjustmensts again for all existing CLusters and 9 new ones locally ... :-))

**Comment by mfucci** [comment]
Yes, let's wait until this is merged with the PR with the new types but otherwise everything seems fine for me.

**Comment by mfucci** [comment]
There seems to be a build issue

**Comment by Apollon77** [comment]
Yes, checking it right now

**Comment by Apollon77** [comment]
PS: Sure it won't ...it bases on UIntegerT :-) fixing now

**Comment by Apollon77** [comment]
@mfucci one question: 

```
Bound(UInt16T,{ min: 0, max: 0xFFFF })
```

or can I remove the max here because it is the UInt16T max itself? So is that "merged" or not?

**Comment by Apollon77** [comment]
@mfucci  types adjusted ... builds now

**Review by mfucci — APPROVED** [review]

File: src/matter/common/VendorId.ts (hunk 11 above)
Comment (Apollon77) [reply src/matter/common/VendorId.ts]: @mfucci Do you have an answer here? (also for future work)

**Pull request merged**